localstack/services/kinesis/kinesis_starter.py

```diff
@@ -21,9 +30,52 @@ def appy_patches():
 
 
 def start_kinesis(port=None, asynchronous=False, update_listener=None):
+    if KINESIS_PROVIDER == 'kinesis-mock':
+        return start_kinesis_mock(port=port, asynchronous=asynchronous, update_listener=update_listener)
+    if KINESIS_PROVIDER == 'kinesalite':
+        return start_kinesalite(port=port, asynchronous=asynchronous, update_listener=update_listener)
+    raise Exception('Unsupported Kinesis provider "%s"' % KINESIS_PROVIDER)
+
+
+def start_kinesis_mock(port=None, asynchronous=False, update_listener=None):
+    target_dir = os.path.join(INSTALL_DIR_INFRA, 'kinesis-mock')
+
+    if platform.startswith('linux'):
```

Comment:
This block will select a more optimized executable than the jar. Average size of these is about 17MB, and it will not need the JVM to execute. 

It falls back on the Jar if this cannot be determined.